### ISSUE TITLE
Need to wrap apk in mac

### ISSUE THREAD

**Issue by lokibyte** [microsoftconnect/intune-app-wrapping-tool-android#56]
could you please provide the wrapping tool for android in mac?

**Comment by bannus** [comment]
Hi @lokibyte, this is not currently supported. However, we are tracking this as a feature request.

(internal tracking #12907503)

**Comment by E-N-G-XOR** [comment]
Use a docker image... works great! you just need to install java and the tools and you are set...  You can bypass the annoying setup installation with an archiving program like winrar...

~~~~~~
FROM mcr.microsoft.com/windows/servercore:ltsc2019-amd64

LABEL Description="Server Core" Vendor="Microsoft" Version="2019"
LABEL Description="Mimigit Porable" Vendor="git" Version="2.42"
LABEL Description="Microsoft Intune App Wrapping Tool for Android" Vendor="Microsoft" Version="latest"
LABEL Description="winrar" Vendor="rarlabs" Version="624"
LABEL Description="Java Jdk" Vendor="oracle" Version="21"
LABEL Description="Android cmdline tools" Vendor="Google" Version="12.0"
LABEL Description="Android build-tools" Vendor="Google" Version="34.0.0"

# NOTE: Must re-wrap after each update or wrap with the latest tools, this happens every 6-8 weeks through GitHub repo ( https://github.com/msintuneappsdk/intune-app-wrapping-tool-android )
# git clone https://github.com/msintuneappsdk/intune-app-wrapping-tool-android.git

# Install git   (https://github.com/git-for-windows/git/releases/)
COPY files/config /
COPY files/installers/MinGit-2.42.0.2-64-bit.zip /minGit.zip
RUN powershell -Command \
#Invoke-WebRequest -Uri "https://github.com/git-for-windows/git/releases/download/v2.42.0.windows.2/MinGit-2.42.0.2-64-bit.zip" -OutFile ".\\minGit.zip" ; \
Expand-Archive -Path ".\\minGit.zip" -DestinationPath 'C:\Program Files\git'; \
Remove-item -Path ".\\minGit.zip" -Force ; \
Move-Item ".\gitconfig" 'C:\Program Files\git\etc' -Force ; \
New-Item -ItemType "Directory" -Path 'C:\Program Files\Microsoft Intune App Wrapping Tool for Android';

# Install Winrar
COPY files/installers/winrar-x64-624.exe /winrar.exe
RUN powershell -Command \
#Invoke-WebRequest -Uri "https://www.win-rar.com/fileadmin/winrar-versions/winrar/winrar-x64-624.exe" -OutFile .\winrar.exe ; \
Start-Process -FilePath .\winrar.exe -ArgumentList '/S' -Wait; \
Remove-item -Path .\winrar.exe ; 

# Install 7Zip
COPY files/installers/7z2301-x64.exe /7z.exe
RUN powershell -Command \
#Invoke-WebRequest -Uri "https://www.7-zip.org/a/7z2301-x64.exe" -OutFile .\7z.exe ; \
Start-Process -FilePath .\7z.exe -ArgumentList '/S' -Wait ; \
Remove-item -Path .\7z.exe ;

# Install Java JDK 21
COPY files/installers/jdk-21_windows-x64_bin.exe /java21.exe
RUN powershell -Command \
# Invoke-WebRequest -Uri "https://download.oracle.com/java/21/latest/jdk-21_windows-x64_bin.exe" -OutFile .\java21.exe ; \
Start-Process -FilePath .\java21.exe -ArgumentList '/s REBOOT=Disable EULA=Disable AUTO_UPDATE=Disable' -Wait ; \
Remove-item -Path .\java21.exe ; 

# Install Android CMDLINE Tools
COPY files/installers/commandlinetools-win-11076708_latest.zip /commandlinetools.zip
RUN powershell -Command \
# Invoke-WebRequest -Uri 'https://dl.google.com/android/repository/commandlinetools-win-11076708_latest.zip' -OutFile '.\\cmdlineTools.zip'; \
Expand-Archive -Path ".\\commandlinetools.zip" -DestinationPath 'C:\Program Files\Android Tools'; \
Remove-item -Path ".\\commandlinetools.zip" -Force ; 

# Update 'env' Variables
RUN setx /M PATH "%PATH%;C:\Program Files\git\cmd;C:\Program Files\Android Tools\cmdline-tools\bin;C:\Program Files\Android Tools\build-tools\34.0.0;C:\Program Files\7-Zip;C:\Program Files\WinRAR;C:\Program Files\git\cmd;C:\Program Files\Microsoft Intune App Wrapping Tool for Android"
RUN setx /M ANDROID_HOME "C:\Program Files\Android Tools" 
RUN setx /M ANDROID_SDK_ROOT "C:\Program Files\Android Tools" 

# Update Android Tools
RUN powershell -Command \
Move-Item ".\licenses" 'C:\Program Files\Android Tools' -Force ; \
Set-Location -Path 'C:\Program Files\Android Tools'; \
sdkmanager --sdk_root='C:\Program Files\Android Tools' --newer ; \
sdkmanager --sdk_root='C:\Program Files\Android Tools' --install 'build-tools;34.0.0';

RUN powershell -Command .\getIntuneTools.ps1

SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'Continue'; $verbosePreference='Continue';"]

**Comment by berkanaslan** [comment]
> 12907503

Could you please provide further details for instructions?

**Comment by E-N-G-XOR** [comment]
Hi, Berkan,
We tried repeatedly to try and get the app to work. In the end we deployed
the app in intune without wrapping it. I talked to our vendor who manages
the app and still had issues with the app crashing on launch. Basically the
app starts, a single sign on routine happens with azure, the the corp logo
is supposed to appear as the app loads data for the session. When the azure
sso happens and the logo appears
.. that is the crash point.
We will have to close/cancel it as un resolved.
Ta,
